<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>485</width>
    <height>401</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>485</width>
    <height>350</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Graph</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="GraphView" name="graphView" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>485</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="file">
    <property name="title">
     <string>Програма</string>
    </property>
    <widget class="QMenu" name="addVertexMenu">
     <property name="title">
      <string>Додати вершини</string>
     </property>
     <addaction name="action_addVericesFromProgram"/>
     <addaction name="action_addVericesFromFile"/>
    </widget>
    <addaction name="addVertexMenu"/>
    <addaction name="action_openSettings"/>
    <addaction name="action_saveGraph"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>Алгоритми</string>
    </property>
    <widget class="QMenu" name="menu">
     <property name="title">
      <string>Пошук в графі</string>
     </property>
     <addaction name="action_breadthSearch"/>
     <addaction name="action_depthSearch"/>
    </widget>
    <widget class="QMenu" name="menu_2">
     <property name="title">
      <string>Вивести матрицю</string>
     </property>
     <addaction name="action_showIncidenceMatrix"/>
     <addaction name="action_showAdjacencyMatrix"/>
     <addaction name="action_showDistanceMatrix"/>
     <addaction name="action_showReachabilityMatrix"/>
     <addaction name="action_showWeightedDistanceMatrix"/>
    </widget>
    <widget class="QMenu" name="menu_4">
     <property name="title">
      <string>Пошук мінімального шляху</string>
     </property>
     <addaction name="action_DijkstraMinPaths"/>
     <addaction name="action_BellmanFord"/>
    </widget>
    <widget class="QMenu" name="menu_5">
     <property name="title">
      <string>Пошук маршутів в графі</string>
     </property>
    </widget>
    <widget class="QMenu" name="menu_6">
     <property name="title">
      <string>Степені вершин</string>
     </property>
     <addaction name="action_showVertexDegree"/>
     <addaction name="action_showHangingVertex"/>
     <addaction name="action_showIsolatedVertex"/>
    </widget>
    <addaction name="menu_2"/>
    <addaction name="menu"/>
    <addaction name="menu_4"/>
    <addaction name="menu_5"/>
    <addaction name="menu_6"/>
    <addaction name="separator"/>
    <addaction name="action_checkCycles"/>
    <addaction name="action_checkConnectivity"/>
    <addaction name="action_topologicalSorting"/>
    <addaction name="action_isPlanarity"/>
    <addaction name="action_coloring"/>
    <addaction name="action_FordFulkerson"/>
    <addaction name="separator"/>
    <addaction name="action_forTest"/>
   </widget>
   <widget class="QMenu" name="menu_7">
    <property name="title">
     <string>Граф</string>
    </property>
    <addaction name="action_inverseGraph"/>
    <addaction name="action_clearScene"/>
   </widget>
   <addaction name="file"/>
   <addaction name="menu_3"/>
   <addaction name="menu_7"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>RightToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_showIncidenceMatrix">
   <property name="text">
    <string>Інцидентності</string>
   </property>
  </action>
  <action name="action_showAdjacencyMatrix">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Суміжності</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="text">
    <string>Вихід</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_showDistanceMatrix">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Відстаней</string>
   </property>
  </action>
  <action name="action_addVericesFromProgram">
   <property name="text">
    <string>Вказати в програмі</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="action_addVericesFromFile">
   <property name="text">
    <string>З файлу</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_showHangingVertex">
   <property name="text">
    <string>Відобразити висячі вершини</string>
   </property>
  </action>
  <action name="action_showVertexDegree">
   <property name="text">
    <string>Відобразити степені вершин</string>
   </property>
  </action>
  <action name="action_openSettings">
   <property name="text">
    <string>Налаштування</string>
   </property>
  </action>
  <action name="action_showIsolatedVertex">
   <property name="text">
    <string>Відобразити ізольовані вершини</string>
   </property>
  </action>
  <action name="action_showReachabilityMatrix">
   <property name="text">
    <string>Досяжності</string>
   </property>
  </action>
  <action name="action_checkCycles">
   <property name="text">
    <string>Знайти деякі цикли в графі</string>
   </property>
  </action>
  <action name="action_breadthSearch">
   <property name="text">
    <string>В ширину</string>
   </property>
  </action>
  <action name="action_depthSearch">
   <property name="text">
    <string>В глибину</string>
   </property>
  </action>
  <action name="action_checkConnectivity">
   <property name="text">
    <string>Визначення зв'язанності графу</string>
   </property>
  </action>
  <action name="action_topologicalSorting">
   <property name="text">
    <string>Топологічне сортування</string>
   </property>
  </action>
  <action name="action_showWeightedDistanceMatrix">
   <property name="text">
    <string>Відстаней(зваж.)</string>
   </property>
  </action>
  <action name="action_DijkstraMinPaths">
   <property name="text">
    <string>Від однієї вершини до всіх інших</string>
   </property>
  </action>
  <action name="action_BellmanFord">
   <property name="text">
    <string>Беллман-Форд</string>
   </property>
  </action>
  <action name="action_isPlanarity">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Перевірити планарність</string>
   </property>
  </action>
  <action name="action_coloring">
   <property name="text">
    <string>Розфарбування графу</string>
   </property>
  </action>
  <action name="action_openWeightGraph">
   <property name="text">
    <string>З файлу(зваж.)</string>
   </property>
  </action>
  <action name="action_saveGraph">
   <property name="text">
    <string>Зберегти</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_clearScene">
   <property name="text">
    <string>Видалити граф</string>
   </property>
  </action>
  <action name="action_FordFulkerson">
   <property name="text">
    <string>Алгоритм Форда — Фалкерсона</string>
   </property>
  </action>
  <action name="action_forTest">
   <property name="text">
    <string>Test</string>
   </property>
  </action>
  <action name="action_inverseGraph">
   <property name="text">
    <string>Інвертувати граф</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GraphView</class>
   <extends>QWidget</extends>
   <header>graphView.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>action_exit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>214</x>
     <y>174</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
